local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,254 @@
+"""Describes obstacles which the Sailbot must avoid: Boats and Land"""
+
+import math
+
+import numpy as np
+from custom_interfaces.msg import HelperAISShip
+from shapely import affinity as af
+from shapely.geometry import Point, Polygon
+
+from local_pathfinding.coord_systems import XY, LatLon, latlon_to_xy, meters_to_km
+
+# Constants
+PROJ_TIME_NO_COLLISION = 3  # hours
+COLLISION_ZONE_SAFETY_BUFFER = 0.5  # km
+COLLISION_ZONE_STRETCH_FACTOR = 1.5  # This factor changes the scope/width of the collision cone
+
+
+class Obstacle:
+    """This class describes general obstacle objects which are
+    anything which the sailbot must avoid.
+
+    Attributes:
+        reference (LatLon): Lat and lon position of the next global waypoint.
+        sailbot_position (XY): Lat and lon position of SailBot.
+        sailbot_speed (float): Speed of the SailBot in kmph.
+        collision_zone (Optional[Polygon]): Shapely polygon representing the
+            obstacle's collision zone. Shape depends on the child class.
+    """
+
+    def __init__(self, reference: LatLon, sailbot_position: LatLon, sailbot_speed: float):
+        self.reference = reference
+        self.sailbot_position_latlon = sailbot_position
+        self.sailbot_position = latlon_to_xy(self.reference, self.sailbot_position_latlon)
+        self.sailbot_speed = sailbot_speed
+
+        # Defined later by the child class
+        self.collision_zone = None
+
+    def is_valid(self, point: XY) -> bool:
+        """Checks if a point is contained the obstacle's collision zone.
+
+        Args:
+            point (LatLon): LatLon Point representing the state point to be checked.
+
+        Returns:
+            bool: True if the point is not within the obstacle's collision zone, false otherwise.
+
+        Raises:
+            ValueError: If the collision zone has not been initialized.
+        """
+        if self.collision_zone is None:
+            raise ValueError("Collision zone has not been initialized")
+
+        # contains() requires a shapely Point object as an argument
+        point = Point(*point)
+
+        return not self.collision_zone.contains(point)
+
+    def update_collision_zone(self, collision_zone: Polygon):
```

Comment:
What do you think about adding what is in `create_collision_zone` here and removing it? This would reduce the number of time we go back and forth between `Boat` and `Obstacle`